engine/common/follower/cache/cache.go

```diff
@@ -159,8 +163,8 @@ func (c *Cache) AddBlocks(batch []*flow.Block) (certifiedBatch []*flow.Block, ce
 	}
 	// caution: in the case `len(batch) == 1`, the `certifiedBatch` might be empty now (else-case)
 
-	// If there exists a parent for the batch's first block, then this is parent is certified by the batch.
-	// Then, we prepend certifiedBatch by the parent
+	// If there exists a parent for the batch's first block, then this is parent is certified
+	//  by the batch. Hence, we prepend certifiedBatch by the parent.
 	if bc.batchParent != nil {
 		s := make([]*flow.Block, 0, 1+len(certifiedBatch))
 		s = append(s, bc.batchParent)
```

Comment:
any particular reason to allocate memory manually? We could just do:
`certifiedBatch = append(bc.batchParent, certifiedBatch...)` pretty sure that's equivalent